x-pack/plugins/security_solution/server/endpoint/routes/metadata/support/agent_status.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClientContract } from 'kibana/server';
+import { AgentService } from '../../../../../../ingest_manager/server';
+import { AgentStatusKueryHelper } from '../../../../../../ingest_manager/common/services';
+import { Agent } from '../../../../../../ingest_manager/common/types/models';
+
+export async function findAgentIDsByStatus(
+  agentService: AgentService,
+  soClient: SavedObjectsClientContract,
+  status: string[],
+  pageSize: number = 1000
+): Promise<string[] | undefined> {
+  const helpers = status.map((s) => {
+    switch (s) {
+      case 'online':
+        return AgentStatusKueryHelper.buildKueryForOnlineAgents();
+      case 'enrolling':
+        return AgentStatusKueryHelper.buildKueryForEnrollingAgents();
+      case 'offline':
+        return AgentStatusKueryHelper.buildKueryForOfflineAgents();
+      case 'error':
+        return AgentStatusKueryHelper.buildKueryForErrorAgents();
+      case 'unenrolling':
+        return AgentStatusKueryHelper.buildKueryForUnenrollingAgents();
+    }
+    return 'INVALID';
```

Comment:
Hopefully we can use the schema to check so we do not have to do this. Such that we will just need to build a KQL query from the array of values passed in.